docs/Paraview.md

```diff
@@ -0,0 +1,62 @@
+# Paraview
+
+## Paraview Web
+
+### Paraview Web Launcher
+
+The paraview web launcher is responsible for starting a new process for each user, generating a session key,
+distributing that session key back to the client, and mapping session key to the port the forked visualization server is running on.
+It listens on localhost:9000.
+
+#### Configuration
+
+The configuration file is located in [../pvw/launcher/config.json](../pvw/launcher/config.json). Here are a few of the most important parameters:
+
+- configuration.sessionURL: This is the url that the paraview launcher expects requests for new sessions to be made to. SESSION_URL_ROOT is replaced with $PROTOCOL://$SERVER_NAME by server.sh. Both $PROTOCOL and $SERVER_NAME are environment variables set by the docker-compose file mentioned in [../README.md](../README.md).
+- configuration.proxyFile: This is the file that will store mappings between sessionIDs and ports that have been opened for the websocket controlled visualization.
+- apps.visualizer: This will run pvpython and is forked for each invocation of the paraview web launcher
+- apps.visualizer-mpi: This provides parallelization for batch operations. I am not sure if this is being used.
+
+#### Logging
+
+`/data/launcher/log/launcherLog.log`
+
+### Paraview Web Visualizer/Server
+
+Each connection will create a new paraview web server process with it's own port.
+The capablities of the visualizer are determined by which VtkWebProtocols are registered to the paraview server.
+An example of one such protocol is the ParaviewWebMouseHandler which automatically handles RPC calls for moving the visualization on the frontend with your mouse.
+Another example is the custom ParaviewWebProtocol, a subclass of vtkWebProtocol, defined in [../pvw/server/app.py](../pvw/server/app.py).
+This is how we define custom actions on the visualization that the frontend can request via RPC calls.
+The VtkWebProtocols are registered with the paraview web server in [../pvw/server/app_server.py](../pvw/server/app_server.py).
+The app_server.py file also configures some of these protocols where necessary.
+
+#### Logging
+
+`/data/launcher/log/<hashed_session_id>.log`
```

Comment:
Can we add something about what is relevant in the logs or when these would be accessed?